src/app/features/increase-fee-drawer/hooks/use-btc-increase-fee.ts

```diff
@@ -0,0 +1,97 @@
+import * as btc from '@scure/btc-signer';
+import { getAddressInfo } from 'bitcoin-address-validation';
+
+import {
+  BitcoinTransaction,
+  BitcoinTransactionVectorOutput,
+} from '@shared/models/transactions/bitcoin-transaction.model';
+import { analytics } from '@shared/utils/analytics';
+
+import { BtcSizeFeeEstimator } from '@app/common/transactions/bitcoin/fees/btc-size-fee-estimator';
+import { useSendFormNavigate } from '@app/pages/send/send-crypto-asset-form/hooks/use-send-form-navigate';
+import { useCurrentNativeSegwitUtxos } from '@app/query/bitcoin/address/utxos-by-address.hooks';
+import { useBitcoinBroadcastTransaction } from '@app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction';
+import { useBitcoinScureLibNetworkConfig } from '@app/store/accounts/blockchain/bitcoin/bitcoin-keychain';
+import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+
+export function useBtcIncreaseFee() {
+  const {
+    address: currentBitcoinAddress,
+    sign,
+    publicKeychain: currentAddressIndexKeychain,
+  } = useCurrentAccountNativeSegwitIndexZeroSigner();
+  const networkMode = useBitcoinScureLibNetworkConfig();
+  const nav = useSendFormNavigate();
+  const { refetch } = useCurrentNativeSegwitUtxos();
+  const { broadcastTx } = useBitcoinBroadcastTransaction();
+
+  function generateTx(payload: { feeRate: string; tx: BitcoinTransaction }) {
+    const newTx = new btc.Transaction();
+    const { vin, vout, fee: prevFee } = payload.tx;
+    const recipient = getRecipientAddress(vout, currentBitcoinAddress) || '';
+    const addressInfo = getAddressInfo(recipient);
+    const p2wpkh = btc.p2wpkh(currentAddressIndexKeychain?.publicKey!, networkMode);
+
+    vin.forEach(input => {
+      newTx.addInput({
+        txid: input.txid,
+        index: input.vout,
+        witnessUtxo: {
+          // script = 0014 + pubKeyHash
+          script: p2wpkh.script,
+          amount: BigInt(input.prevout.value),
+        },
+      });
+    });
+
+    const txSizer = new BtcSizeFeeEstimator();
+    const sizeInfo = txSizer.calcTxSize({
+      // Only p2wpkh is supported by the wallet
+      input_script: 'p2wpkh',
+      input_count: vin.length,
+      // From the address of the recipient, we infer the output type
+      [addressInfo.type + '_output_count']: 2,
+    });
```

Comment:
Wonder if we could better package this as a fn accepting these 3 params? My code from original btc implementation, but is messy and repeated in many places.